### ISSUE TITLE
Encountered error when running on real world data

### ISSUE THREAD

**Issue by pspealman** [jszavits/NEAR#6]
Hi,

I'm encountering an error when running this on my data (see attached) but it works without error on the demo data. The error I get is:

`./NEAR
double free or corruption (!prev)

Program received signal SIGABRT: Process abort signal.

Backtrace for this error:
#0  0x1494d0f8d171 in ???
#1  0x1494d0f8c313 in ???
#2  0x1494d041cb1f in ???
#3  0x1494d041ca9f in ???
#4  0x1494d03efe04 in ???
#5  0x1494d045f036 in ???
#6  0x1494d046619b in ???
#7  0x1494d0467f4b in ???
#8  0x1494d1188d8d in ???
#9  0x1494d119b256 in ???
#10  0x405adc in ???
#11  0x40bf46 in ???
#12  0x1494d0408cf2 in ???
#13  0x4010ad in ???
#14  0xffffffffffffffff in ???
Aborted (core dumped)`

I've attached a zip with my A-site, .dat, and log file. The A-site and .dat are formatted the same as the demo files. 
The log file suggests I'm only providing a single codon long gene but I'm not clear as to why. 
[near_issue.zip](https://github.com/jszavits/NEAR/files/13258130/near_issue.zip)

**Comment by jszavits** [comment]
Hi,

in the genelist.dat you have put the gene name "YKL039W", however you have attached the file A-site_YAL003W.dat. Can you try putting the correct name YAL003W in the genelist.dat or attach A-site_YKL039W.dat for me to inspect the file?

**Comment by pspealman** [comment]
Thanks for the rapid reply! My apologies I bundled the wrong file - I've attached the correct .dat
[near_issue.zip](https://github.com/jszavits/NEAR/files/13259664/near_issue.zip)

**Comment by jszavits** [comment]
Thanks. I took a while, but I got where the error comes from. The error appears when your genename cannot be matched to the corresponding A-site file in the ribo-seq folder. In that case, the program will still try to open the non-existent file and will think that there are 0 codons (without start codon)+ start codon = 1. 

I've updated the main program so that it will let you know (in the log file) if it cannot find the A-site file, in which case this gene will be skipped.
